run_xnat2bids.py

```diff
@@ -331,15 +362,100 @@ async def main():
             "command": sbatch_cmd_without_password
         })
         
-        # Run xnat2bids asynchronously.
-        task = asyncio.create_task(asyncio.create_subprocess_exec(*sbatch_cmd))
-        tasks.append(task)
+        # Run xnat2bids 
+        proc = await asyncio.create_subprocess_exec(*sbatch_cmd, stdout=asyncio.subprocess.PIPE)
+
+        stdout, stderr = await proc.communicate()
+        output.append(stdout)
+    
+    return output
+
+async def launch_bids_validator(arg_dict, user, password, bids_root, job_deps):    
+
+    # Establish connection 
+    connection = requests.Session()
+    connection.verify = True
+    connection.auth = (user, password)
+
+    # Fetch pi and study prefixes for BIDS path
+    host = arg_dict["xnat2bids-args"]["host"]
+    session = arg_dict["xnat2bids-args"]["sessions"][0]
+    proj, subj = get_project_subject_session(connection, host, session)
+    pi_prefix, study_prefix = prepare_path_prefixes(proj, subj)
+
+    # Define bids-validator singularity image path
+    simg=f"/gpfs/data/bnc/simgs/bids/validator-latest.sif"
+
+    # Define bids_experiment_dir
+    bids_experiment_dir = f"{bids_root}/{pi_prefix}/{study_prefix}/bids"
+
+    # Build shell script for sbatch
+    sbatch_bids_val_script = f"\"$(cat << EOF #!/bin/sh\n \
+        apptainer exec --no-home -B {bids_experiment_dir} {simg} \
+        bids-validator {bids_experiment_dir}\nEOF\n)\""
+
+    # Compile list of slurm parameters.
+    bids_val_slurm_params = compile_slurm_list(arg_dict, user)
+    output = f"/gpfs/scratch/{user}/logs/%x-%J.txt"
+    arg = f"--output {output}"
+    bids_val_slurm_params.append(arg)
+    slurm_options = ' '.join(bids_val_slurm_params)
+
+    # Process command string for SRUN
+    slurm_options = slurm_options.replace("--job-name xnat2bids", "--job-name bids-validator")
+
+    # Fetch JOB-IDs of xnat2bids jobs to wait upon
+    afterok_ids = ":".join(job_deps)
+
+    sbatch_bids_val_cmd = shlex.split(f"sbatch -d afterok:{afterok_ids} {slurm_options} \
+        --wrap {sbatch_bids_val_script}") 
+
+    # Run bids-validator
+    proc = await asyncio.create_subprocess_exec(*sbatch_bids_val_cmd, stdout=asyncio.subprocess.PIPE)
```

Comment:
we should make sure that this validation step doesn't get run if they used the export-only flag

Reply:
Definitely!  Great catch, thank you.  I'm returning `needs_validation` from `launch_x2b_jobs` if **_at least one_** of the argument lists does not contain `--export-only`

```
  x2b_output, needs_validation = await launch_x2b_jobs(argument_lists, simg)
  update_jobs(jobs, x2b_output)

  # Launch bids-validator
  if needs_validation:
      validator_output = await launch_bids_validator(arg_dict, user, password, bids_root, jobs)
      update_jobs(jobs, validator_output)
```